src/types/polygon.rs

```diff
@@ -70,6 +72,58 @@ where
     }
 }
 
+impl<T: WktNum> PolygonTrait for Polygon<T> {
+    type T = T;
+    type RingType<'a> = &'a LineString<T> where Self: 'a;
+
+    fn dim(&self) -> geo_traits::Dimensions {
+        // TODO: infer dimension from empty WKT
+        if self.0.is_empty() {
+            geo_traits::Dimensions::Xy
+        } else {
+            self.0[0].dim()
+        }
+    }
+
+    fn exterior(&self) -> Option<Self::RingType<'_>> {
+        self.0.first()
+    }
+
+    fn num_interiors(&self) -> usize {
+        self.0.len().saturating_sub(1)
+    }
+
+    unsafe fn interior_unchecked(&self, i: usize) -> Self::RingType<'_> {
+        &self.0[i + 1]
+    }
+}
+
+impl<T: WktNum> PolygonTrait for &Polygon<T> {
+    type T = T;
+    type RingType<'a> = &'a LineString<T> where Self: 'a;
+
+    fn dim(&self) -> geo_traits::Dimensions {
+        // TODO: infer dimension from empty WKT
+        if self.0.is_empty() {
+            geo_traits::Dimensions::Xy
+        } else {
+            self.0[0].dim()
+        }
+    }
+
+    fn exterior(&self) -> Option<Self::RingType<'_>> {
+        self.0.first()
+    }
+
+    fn num_interiors(&self) -> usize {
+        self.0.len().saturating_sub(1)
+    }
+
+    unsafe fn interior_unchecked(&self, i: usize) -> Self::RingType<'_> {
+        &self.0[i + 1]
```

Comment:
Based on my latest understanding after https://github.com/georust/geojson/pull/245/files#r1821255299,

Other than the `CoordTrait.nth_unchecked`, shouldn't these `unchecked` methods be using https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get_unchecked like https://github.com/georust/geojson/pull/245?